diff -ru nss-3.12.10/mozilla/security/nss/lib/ckfw/builtins/ckbiver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/builtins/ckbiver.c
--- nss-3.12.10/mozilla/security/nss/lib/ckfw/builtins/ckbiver.c	2004-08-31 19:28:50.000000000 +0200
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/builtins/ckbiver.c	2011-11-25 20:35:27.536298403 +0100
@@ -52,7 +52,7 @@
  */
 const char __nss_builtins_rcsid[] = "$Header: NSS Builtin Trusted Root CAs "
         NSS_BUILTINS_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_builtins_sccsid[] = "@(#)NSS Builtin Trusted Root CAs "
         NSS_BUILTINS_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/ckfw/capi/ckcapiver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/capi/ckcapiver.c
--- nss-3.12.10/mozilla/security/nss/lib/ckfw/capi/ckcapiver.c	2005-11-04 03:05:04.000000000 +0100
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/capi/ckcapiver.c	2011-11-25 20:35:27.543298349 +0100
@@ -53,7 +53,7 @@
  */
 const char __nss_ckcapi_rcsid[] = "$Header: NSS Access to Microsoft Certificate Store "
         NSS_CKCAPI_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_ckcapi_sccsid[] = "@(#)NSS Access to Microsoft Certificate Store "
         NSS_CKCAPI_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/ckfw/nssmkey/ckmkver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/nssmkey/ckmkver.c
--- nss-3.12.10/mozilla/security/nss/lib/ckfw/nssmkey/ckmkver.c	2005-11-24 00:04:08.000000000 +0100
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/nssmkey/ckmkver.c	2011-11-25 20:35:27.584298033 +0100
@@ -53,7 +53,7 @@
  */
 const char __nss_ckmk_rcsid[] = "$Header: NSS Access to the MAC OS X Key Ring "
         NSS_CKMK_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_ckmk_sccsid[] = "@(#)NSS Access to the MAC OS X Key Ring "
         NSS_CKMK_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/ckfw/pem/ckpemver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/pem/ckpemver.c
--- nss-3.12.10/mozilla/security/nss/lib/ckfw/pem/ckpemver.c	2010-11-25 19:49:27.000000000 +0100
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/ckfw/pem/ckpemver.c	2011-11-25 20:35:27.611297824 +0100
@@ -53,7 +53,7 @@
  */
 const char __nss_ckpem_rcsid[] = "$Header: NSS Access to Flat Files in PEM format"
         NSS_CKPEM_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_ckcapi_sccsid[] = "@(#)NSS Access to Flag Files in PEM format "
         NSS_CKPEM_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/freebl/freeblver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/freebl/freeblver.c
--- nss-3.12.10/mozilla/security/nss/lib/freebl/freeblver.c	2006-12-05 00:31:35.000000000 +0100
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/freebl/freeblver.c	2011-11-25 20:35:12.541413873 +0100
@@ -51,6 +51,6 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_freebl_rcsid[] = "$Header: NSS " SOFTOKEN_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_freebl_sccsid[] = "@(#)NSS " SOFTOKEN_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/nss/nssver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/nss/nssver.c
--- nss-3.12.10/mozilla/security/nss/lib/nss/nssver.c	2004-04-25 17:03:12.000000000 +0200
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/nss/nssver.c	2011-11-25 20:35:12.660412957 +0100
@@ -51,6 +51,6 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_base_rcsid[] = "$Header: NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_base_sccsid[] = "@(#)NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/smime/smimever.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/smime/smimever.c
--- nss-3.12.10/mozilla/security/nss/lib/smime/smimever.c	2004-04-25 17:03:16.000000000 +0200
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/smime/smimever.c	2011-11-25 20:35:13.069409807 +0100
@@ -51,6 +51,6 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_smime_rcsid[] = "$Header: NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_smime_sccsid[] = "@(#)NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/softoken/softkver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/softoken/softkver.c
--- nss-3.12.10/mozilla/security/nss/lib/softoken/softkver.c	2006-12-05 00:31:35.000000000 +0100
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/softoken/softkver.c	2011-11-25 20:35:13.248408430 +0100
@@ -51,6 +51,6 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_softokn_rcsid[] = "$Header: NSS " SOFTOKEN_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_softokn_sccsid[] = "@(#)NSS " SOFTOKEN_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/ssl/sslver.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/ssl/sslver.c
--- nss-3.12.10/mozilla/security/nss/lib/ssl/sslver.c	2004-04-25 17:03:17.000000000 +0200
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/ssl/sslver.c	2011-11-25 20:35:14.066402131 +0100
@@ -51,6 +51,6 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_ssl_rcsid[] = "$Header: NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_ssl_sccsid[] = "@(#)NSS " NSS_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
diff -ru nss-3.12.10/mozilla/security/nss/lib/util/secoid.c nss-3.12.10-notimestamps/mozilla/security/nss/lib/util/secoid.c
--- nss-3.12.10/mozilla/security/nss/lib/util/secoid.c	2010-08-16 20:28:39.000000000 +0200
+++ nss-3.12.10-notimestamps/mozilla/security/nss/lib/util/secoid.c	2011-11-25 20:35:14.162401391 +0100
@@ -59,9 +59,9 @@
  * must not end in a '$' to prevent rcs keyword substitution.
  */
 const char __nss_util_rcsid[] = "$Header: NSS " NSSUTIL_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_util_sccsid[] = "@(#)NSS " NSSUTIL_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
 
 /* MISSI Mosaic Object ID space */
 #define USGOV                   0x60, 0x86, 0x48, 0x01, 0x65
